src/main/java/domain/model/Goods.java

```
@@ -0,0 +1,35 @@
+package domain.model;
+
+import domain.vo.Name;
+
+import java.util.List;
+
+public class Goods {
+    private static final String NOT_SAME_COUNT_ERROR = "참가자 수와 상품 수가 같아야 합니다.";
+    private final List<Name> items;
+    private final int playerCounts;
+
+    public Goods(List<Name> items, int playerCounts) {
+        this.items = items;
+        this.playerCounts = playerCounts;
+        validate();
```

값을 할당하기 전에 검증을 먼저하는건 어떨까요?
fail fast라는 키워드로 검색해보세요 :)
인자값이 잘못되었을 경우 할당을 하고 검증 로직을 탈 이유는 없으니, 먼저 검증하고 그 후 정상적인 값의 경우에만 할당하는 방법입니다. 이는 잘못된 값임에도 불구하고 계속 실행해 코드의 다른곳에서 문제를 일으키는 상황을 방지하기 위해 사용되는 기법입니다. 단순 생성자 인자 검증 뿐 아니라 서비스 로직에서도 사용됩니다.